From c1caaa013c6b3fb1ff3c0fafe7013ef40b369d75 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@nxp.com>
Date: Fri, 28 Jul 2017 17:11:12 +0800
Subject: [PATCH 2309/5242] MLK-16013-30 staging: typec: Reset
 hard_reset_count to be 0 after send source cap

commit  5217d2cb3ae2a8fb2ebadd1c693dc37e613189f5 from
https://source.codeaurora.org/external/imx/linux-imx.git

Per standard, we should clear the hard reset count to be 0 after souce
sent the source cap.

Reviewed-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/typec/tcpm.c |    7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/drivers/usb/typec/tcpm.c b/drivers/usb/typec/tcpm.c
index 6da8a61..7f88944 100644
--- a/drivers/usb/typec/tcpm.c
+++ b/drivers/usb/typec/tcpm.c
@@ -2938,12 +2938,7 @@ static void run_state_machine(struct tcpm_port *port)
 			tcpm_set_state(port, SRC_SEND_CAPABILITIES,
 				       PD_T_SEND_SOURCE_CAP);
 		} else {
-			/*
-			 * Per standard, we should clear the reset counter here.
-			 * However, that can result in state machine hang-ups.
-			 * Reset it only in READY state to improve stability.
-			 */
-			/* port->hard_reset_count = 0; */
+			port->hard_reset_count = 0;
 			port->caps_count = 0;
 			port->pd_capable = true;
 			hrtimer_start(&port->snd_res_timer,
-- 
1.7.9.5

